### PR TITLE
Skip requesting genesis block in PeerChainValidator

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
When validating that a peer is on the same chain, if we haven't finalised anything past genesis, assume that the fork digest check is enough to validate the chain and don't specifically request genesis.

### CODE REVIEW

File: networking/eth2/src/test/java/tech/pegasys/artemis/networking/eth2/peers/PeerChainValidatorTest.java [hunk 1]
```diff
@@ -317,15 +317,18 @@ private void remoteChainIsAheadOnSameChain() {
     when(peer.requestBlockBySlot(earlierBlockSlot)).thenReturn(blockFuture);
   }
 
-  private void remoteOnSameChainButWillNotReturnGenesis() {
-    final SafeFuture<SignedBeaconBlock> postGenesisBlock = SafeFuture.completedFuture(earlierBlock);
+  private void remoteOnSameChainButReturnsResponseWhenGenesisRequested() {
```
Comment (mbaxter) [inline networking/eth2/src/test/java/tech/pegasys/artemis/networking/eth2/peers/PeerChainValidatorTest.java]:
```suggestion
  private void remoteOnSameChainButReturnsInvalidResponseWhenGenesisRequested() {
```

File: networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/peers/PeerChainValidator.java [hunk 2]
```diff
@@ -193,11 +193,14 @@ private boolean remoteEpochIsInvalid(
         .thenApply(maybeBlock -> blockToSlot(finalizedEpochSlot, maybeBlock))
         .thenCompose(
             blockSlot -> {
+              if (blockSlot.equals(UnsignedLong.valueOf(Constants.GENESIS_SLOT))) {
```
Comment (mbaxter) [inline networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/peers/PeerChainValidator.java:201]: I think we can return early before we pull this block by checking if the `finalizedEpochSlot` is the genesis slot

**Review by mbaxter — APPROVED** [review]

File: networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/peers/PeerChainValidator.java (hunk 2 above)
Comment (ajsutton) [reply networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/peers/PeerChainValidator.java:201]: We can but we need the existing check as well.  We might have finalised epoch 1 without any blocks ever being produced, so the block in effect would still be the genesis block.

File: networking/eth2/src/test/java/tech/pegasys/artemis/networking/eth2/peers/PeerChainValidatorTest.java (hunk 1 above)
Comment (ajsutton) [reply networking/eth2/src/test/java/tech/pegasys/artemis/networking/eth2/peers/PeerChainValidatorTest.java]: Done.

**Pull request merged**